src/node/http-request.ts

```diff
@@ -0,0 +1,33 @@
+import { HttpRequest, HttpMethod, HttpBody, HttpRequestHeaders } from "../http";
+
+export class HttpRequestImpl implements HttpRequest {
```

Comment:
I think we can move this file outside of `node` folder.

Reply:
I agree

Reply:
I have decided to keep this in the node folder. I assume that you want to move this out because we want to use this in Deno as well. In order to do that we have to stay away from NPM Packages. At this phase of the project, I believe that we should create something that people can try out ASAP and we can refactor the code later as long as we have well-designed interfaces, otherwise implementing all the HTTP utilities without the help of the libraries I have used seems to be quite a time-consuming process.